### ISSUE TITLE
chevreul

### ISSUE THREAD

**Issue by whtns** [Bioconductor/Contributions#3332]
Update the following URL to point to the GitHub repository of
the package you wish to submit to _Bioconductor_

- Repository: https://github.com/whtns/chevreul/

Confirm the following by editing each check box to '[x]'

- [x] I understand that by submitting my package to _Bioconductor_,
  the package source and all review commentary are visible to the
  general public.

- [x] I have read the _Bioconductor_ [Package Submission][2]
  instructions. My package is consistent with the _Bioconductor_
  [Package Guidelines][1].

- [x] I understand Bioconductor [Package Naming Policy][9] and acknowledge
  Bioconductor may retain use of package name.

- [x] I understand that a minimum requirement for package acceptance 
  is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS. 
  Passing these checks does not result in automatic acceptance. The 
  package will then undergo a formal review and recommendations for 
  acceptance regarding other Bioconductor standards will be addressed.

- [x] My package addresses statistical or bioinformatic issues related
  to the analysis and comprehension of high throughput genomic data.

- [x] I am committed to the long-term maintenance of my package. This
  includes monitoring the [support site][3] for issues that users may
  have, subscribing to the [bioc-devel][4] mailing list to stay aware
  of developments in the _Bioconductor_ community, responding promptly
  to requests for updates from the Core team in response to changes in
  _R_ or underlying software.
  
- [x] I am familiar with the [Bioconductor code of conduct][7] and 
  agree to abide by it.

I am familiar with the essential aspects of _Bioconductor_ software
management, including:

- [x] The 'devel' branch for new packages and features.
- [x] The stable 'release' branch, made available every six
      months, for bug fixes.
- [x] _Bioconductor_ version control using [Git][5]
  (optionally [via GitHub][6]).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website][8] to sign up.

[1]: https://contributions.bioconductor.org/
[2]: https://bioconductor.org/developers/package-submission/
[3]: https://support.bioconductor.org
[4]: https://stat.ethz.ch/mailman/listinfo/bioc-devel
[5]: http://bioconductor.org/developers/how-to/git/
[6]: http://bioconductor.org/developers/how-to/git/sync-existing-repositories/
[7]: https://bioconductor.org/about/code-of-conduct/
[8]: https://bioconductor.org/
[9]: https://bioconductor.org/developers/package-submission/#naming

**Comment by lshep** [comment]
All package dependencies must be on CRAN or Bioconductor. `annotables` is currently not on CRAN or Bioconductor and must not be used or referenced in the package.  

I assume `chevreuldata` will be submitted once this issue can accept additional packages.   I glanced at that github repo. All Bioconductor packages must include a vignette.

**Comment by whtns** [comment]
Thank you for taking a look at these packages. I removed the annotables dependency in chevreul and added a vignette in chevreuldata. I'd greatly appreciate further review.

**Comment by whtns** [comment]
adding an explicit reference to accompanying data package `chevreuldata`

 AdditionalPackage: https://github.com/whtns/chevreuldata

**Comment by lshep** [comment]
A few questions: 

Can default for saving or writing files be a tempdir() unless specified by a user?

Insteading of defining `~/.cache/` which isn't necessarily a default for all operating systems can you use `tools::R_user_dir(which="cache")` or `rappdirs::user_cache_dir` which would grab an appropriate path for a given OS.

Why is a `chmod` explicit set_permission_call necessary?

It looks like you import httr::GET but is it utilized anywhere?

**Comment by whtns** [comment]
Thank you for continuing to share feedback/shortcomings. I addressed the points you raised above in my last commit. 

_Can default for saving or writing files be a tempdir() unless specified by a user?_
I've removed any hardcoded filepaths previously used for saving

Insteading of defining ~/.cache/ which isn't necessarily a default for all operating systems can you use tools::R_user_dir(which="cache") or rappdirs::user_cache_dir which would grab an appropriate path for a given OS.
I've replaced references to caches with rappdirs::user_cache_dir

Why is a chmod explicit set_permission_call necessary?
removed

It looks like you import httr::GET but is it utilized anywhere?
removed

**Comment by lshep** [comment]
When I try to run R CMD build I'm getting the following 
```
 R CMD build chevreul
* checking for file 'chevreul/DESCRIPTION' ... OK
* preparing 'chevreul':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
--- re-building ‘chevreul.Rmd’ using rmarkdown
--- finished re-building ‘chevreul.Rmd’

--- re-building ‘integration.Rmd’ using rmarkdown

Quitting from lines  at lines 71-77 [unnamed-chunk-2] (integration.Rmd)
Error: processing vignette 'integration.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘integration.Rmd’

--- re-building ‘preprocessing.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
--- finished re-building ‘preprocessing.Rmd’

--- re-building ‘shiny_app.Rmd’ using rmarkdown
--- finished re-building ‘shiny_app.Rmd’

--- re-building ‘visualization.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page

Quitting from lines  at lines 144-145 [unnamed-chunk-5] (visualization.Rmd)
Error: processing vignette 'visualization.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘visualization.Rmd’

SUMMARY: processing the following files failed:
  ‘integration.Rmd’ ‘visualization.Rmd’

Error: Vignette re-building failed.

```

It looks like you are hard linking to a file on your system that would not exist for a user.

**Comment by whtns** [comment]
I'm sorry I can't identify the source of the errors listed above in my current branch. I incremented the package version. Can you confirm that the check and build is run against the devel branch? 
thanks,

**Comment by lshep** [comment]
Yes it is 
```
lorikern@jbcj433:~/PkgReview/PreReview/chevreul(devel)$ git remote -v
origin	https://github.com/whtns/chevreul/ (fetch)
origin	https://github.com/whtns/chevreul/ (push)


lorikern@jbcj433:~/PkgReview/PreReview/chevreul(devel)$ git pull
remote: Enumerating objects: 7, done.
remote: Counting objects: 100% (7/7), done.
remote: Compressing objects: 100% (2/2), done.
remote: Total 7 (delta 5), reused 7 (delta 5), pack-reused 0
Unpacking objects: 100% (7/7), 620 bytes | 38.00 KiB/s, done.
From https://github.com/whtns/chevreul
   e4d2858..da02a9c  sce        -> origin/sce
Already up to date.


lorikern@jbcj433:~/PkgReview/PreReview/chevreul(devel)$ cd ..
lorikern@jbcj433:~/PkgReview/PreReview$ R CMD build chevreul
* checking for file 'chevreul/DESCRIPTION' ... OK
* preparing 'chevreul':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
--- re-building ‘chevreul.Rmd’ using rmarkdown
--- finished re-building ‘chevreul.Rmd’

--- re-building ‘integration.Rmd’ using rmarkdown

Quitting from lines 71-77 [unnamed-chunk-2] (integration.Rmd)
Error: processing vignette 'integration.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘integration.Rmd’

--- re-building ‘preprocessing.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
--- finished re-building ‘preprocessing.Rmd’

--- re-building ‘shiny_app.Rmd’ using rmarkdown
--- finished re-building ‘shiny_app.Rmd’

--- re-building ‘visualization.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page

Quitting from lines 144-145 [unnamed-chunk-5] (visualization.Rmd)
Error: processing vignette 'visualization.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘visualization.Rmd’

SUMMARY: processing the following files failed:
  ‘integration.Rmd’ ‘visualization.Rmd’

Error: Vignette re-building failed.
Execution halted

```

**Comment by whtns** [comment]
Ah, thank you very much. I had a mismatch between local and remote branches. The branch you are pulling is origin/sce in the above snippet. I've updated the local and remote branches and incremented version again. Can you try building again? Sorry for the trouble. 

```
skevin@cobrinik-srt:~/rpkgs/chevreul$ git status
On branch devel
Changes to be committed:
  (use "git restore --staged <file>..." to unstage)
        deleted:    docs/articles/Heatmap2.0.PNG
        deleted:    docs/articles/PlotCoverage2.0.PNG
        deleted:    docs/articles/violinPlot.PNG

skevin@cobrinik-srt:~/rpkgs/chevreul$ ..
skevin@cobrinik-srt:~/rpkgs$ R CMD build chevreul
* checking for file 'chevreul/DESCRIPTION' ... OK
* preparing 'chevreul':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... OK
* checking for LF line-endings in source and make files and shell scripts
* checking for empty or unneeded directories
* looking to see if a 'data/datalist' file should be added
* building 'chevreul_0.99.2.tar.gz'
```

**Comment by lshep** [comment]
Same 
```
R CMD build chevreul
* checking for file 'chevreul/DESCRIPTION' ... OK
* preparing 'chevreul':
* checking DESCRIPTION meta-information ... OK
* installing the package to build vignettes
* creating vignettes ... ERROR
--- re-building ‘chevreul.Rmd’ using rmarkdown
--- finished re-building ‘chevreul.Rmd’

--- re-building ‘integration.Rmd’ using rmarkdown

Quitting from lines 71-77 [unnamed-chunk-2] (integration.Rmd)
Error: processing vignette 'integration.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘integration.Rmd’

--- re-building ‘preprocessing.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
--- finished re-building ‘preprocessing.Rmd’

--- re-building ‘shiny_app.Rmd’ using rmarkdown
--- finished re-building ‘shiny_app.Rmd’

--- re-building ‘visualization.Rmd’ using rmarkdown
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page
Warning in grid.Call.graphics(C_points, x$x, x$y, x$pch, x$size) :
  semi-transparency is not supported on this device: reported only once per page

Quitting from lines 144-145 [unnamed-chunk-5] (visualization.Rmd)
Error: processing vignette 'visualization.Rmd' failed with diagnostics:
file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist
--- failed re-building ‘visualization.Rmd’

SUMMARY: processing the following files failed:
  ‘integration.Rmd’ ‘visualization.Rmd’

Error: Vignette re-building failed.
Execution halted

```

In the vignette this seems to have to do with the plot_violin function.  For what its worth it is also reproducible if I use devtools to load the package and just try to run that example as well 
```
> load_all(".")
ℹ Loading chevreul
> chevreul_sce <- chevreuldata::human_gene_transcript_sce()
snapshotDate(): 2024-05-28
see ?chevreuldata and browseVignettes('chevreuldata') for documentation
loading from cache
> plot_violin(chevreul_sce, plot_var = "batch", features = "NRL")
Error in .Call2("C_h5getdimscales", filepath, name, scalename, PACKAGE = "HDF5Array") : 
  file '/dataVolume/storage/single_cell_projects/sc_cone_devel/sc_cone_devel_organoid/20181001-DS-organoid-Hs_proj/output/velocyto/20181001-DS-organoid-Hs_proj.loom' does not exist

```

**Comment by whtns** [comment]
Ok, I think I've resolved this problem. The example dataset (in accompanying experimenthub package chevreuldata) includes a SingleCellLoomExperiment assay for use with the velociraptor package. Because the source loom file is inaccessible on any other system even unrelated functions break. Because I don't know best practices to share loom files in this manner I've removed that assay from the dataset and excluded those functions from vignettes and examples.

**Comment by lshep** [comment]
You should not disable these examples and vignettes. If the data isn't loading, you should check out LoomExperiment where this class is defined and figure out the proper way.  Exported functions need to be run/tested somewhere in the code base.

**Comment by whtns** [comment]
Ok thanks, I recognize the submission requirement to have included functions documented with examples. I've removed any dependency on LoomExperiment or velociraptor from the package

**Comment by lshep** [comment]
monocle3 does not appear to be on CRAN or Bioconductor.  Bioconductor packages require all dependencies to be on CRAN or Bioconductor.  You must remove the dependency from your package or request the monocle3 maintainers to officially submit the package.

**Comment by whtns** [comment]
Thank you, we've removed all references to monocle3 and dropped the package dependency in DESCRIPTION

**Comment by whtns** [comment]
I am submitting an additional experimentHub data package, chevreuldata. See #3491. I'm also pinging author @Bavanaa. This is a dependency for the above package chevreul  

Update the following URL to point to the GitHub repository of
the package you wish to submit to Bioconductor

Repository: https://github.com/whtns/chevreuldata

Confirm the following by editing each check box to '[x]'

[ x] I understand that by submitting my package to Bioconductor,
the package source and all review commentary are visible to the
general public.

[x ] I have read the Bioconductor [Package Submission](https://bioconductor.org/developers/package-submission/)
instructions. My package is consistent with the Bioconductor
[Package Guidelines](https://contributions.bioconductor.org/).

[x ] I understand Bioconductor [Package Naming Policy](https://bioconductor.org/developers/package-submission/#naming) and acknowledge
Bioconductor may retain use of package name.

[ x] I understand that a minimum requirement for package acceptance
is to pass R CMD check and R CMD BiocCheck with no ERROR or WARNINGS.
Passing these checks does not result in automatic acceptance. The
package will then undergo a formal review and recommendations for
acceptance regarding other Bioconductor standards will be addressed.

[ x] My package addresses statistical or bioinformatic issues related
to the analysis and comprehension of high throughput genomic data.

[x ] I am committed to the long-term maintenance of my package. This
includes monitoring the [support site](https://support.bioconductor.org/) for issues that users may
have, subscribing to the [bioc-devel](https://stat.ethz.ch/mailman/listinfo/bioc-devel) mailing list to stay aware
of developments in the Bioconductor community, responding promptly
to requests for updates from the Core team in response to changes in
R or underlying software.

[x ] I am familiar with the [Bioconductor code of conduct](https://bioconductor.org/about/code-of-conduct/) and
agree to abide by it.

I am familiar with the essential aspects of Bioconductor software
management, including:

[ x] The 'devel' branch for new packages and features.
[ x] The stable 'release' branch, made available every six
months, for bug fixes.
[ x] Bioconductor version control using [Git](http://bioconductor.org/developers/how-to/git/)
(optionally [via GitHub](http://bioconductor.org/developers/how-to/git/sync-existing-repositories/)).

For questions/help about the submission process, including questions about
the output of the automatic reports generated by the SPB (Single Package
Builder), please use the #package-submission channel of our Community Slack.
Follow the link on the home page of the [Bioconductor website](https://bioconductor.org/) to sign up.

**Comment by lshep** [comment]
Please see  https://github.com/Bioconductor/Contributions#submitting-related-packages  and use ` AdditionalPackage: https://github.com/whtns/chevreuldata`

**Comment by whtns** [comment]
AdditionalPackage: https://github.com/whtns/chevreuldata

**Comment by whtns** [comment]
Hi @lshep, I'm continuing discussions from community-bioc slack and experimenthub github issues. I have two concerns in my experimenthub and conventional package submissions `chevreuldata` and `chevreul`: 

1. unable to build `chevreuldata` with new versions; with an accompanying error message in the build reports [ex.](https://bioconductor.org/spb_reports/chevreuldata_buildreport_20240814124720.html) I am able to install the package from on my local machine and access resources without issue. I am having trouble interpreting the build reports to debug
2. large files present in upstream git@git.bioconductor.org:packages/chevreul 
http://contributions.bioconductor.org/git-version-control.html#new-package-workflow
https://contributions.bioconductor.org/git-version-control.html#remove-large-data-files-and-clean-git-tree
I run git filter-repo to remove them from my github repository but when i try to push upstream I am rejected because I haven’t pulled the large files from the upstream [git.bioconductor.org](http://git.bioconductor.org/)

**Comment by lshep** [comment]
I'll reset chevreul now and kick off a new build.    The chevreuldata is interesting as when I go on the builder I can access everything when I run the code chunk manually and when I build/check.  I'll kick off a new build of that as well when I'm done but otherwise will need to investigate more

**Comment by lshep** [comment]
this seems to be an issue on our end.  We are investigating

**Comment by LiNk-NY** [comment]
Hi @whtns 
Thank you for your submission. 
Please see the review below. 

Best regards,
Marcel

--- 

# chevreul #3332

* Overall the package contains a lot of functionality and as a consequence,
dependencies. It would be better to split the package into possibly two or
three separate packages that minimize the number of dependencies in each.
* Consider the `alabaster` framework for exporting `SingleCellExperiment`
and other objects in a language-agnostic format.

## DESCRIPTION

* Is it possible to cut down some of the dependencies? The number of imports
can make the package fragile.
* You may also want to split the package into separate packages (e.g., shiny
and visualization packages)

## NAMESPACE

* Looks good.
* Consider removing the `%>%` dependency.
* Use more descriptive function names, e.g., `plot_feature`, `plot_readcount`,
`plot_var`, etc.

## vignettes

* Use more descriptive titles in the vignette / index entry.
* Remove `Sys.time` calls from vignette. These will be provided by the BBS.
* (optional) Consider wrapping text longer than 80 column width.
* Although a good reminder, it is not necessary to display
`BiocManager::valid()` for the user.
* The `human_gene_transcript_sce()` function in `chevreuldata` does not seem to
be working. I get some "resource removed" errors.
* The code is a bit unclear. You take the `chevreul_sce` and reconstruct
another `SingleCellExperiment`?
* The code for creating the vignette is not needed. Please remove.
* Avoid static images of plots and provide the code needed to plot. Consider
moving plotting functionality into a separate package.
* Include code to reproduce plots and analyses. Do not use static images.

## R

* Properly reference data objects in `transcripts_to_genes` etc. as detailed
in: https://contributions.bioconductor.org/data.html?q=data#exported-data-and-the-data-directory
* Minimize the code repitition by extracting the common operations in both
conditions, e.g. in `chevreul_app.R` where
`featureType %in% c("transcript", "gene")`
* Based on the `shiny` docs, `moduleServer()` is preferred over `callModule`.
Please update accordingly.
* Be more descriptive in the documentation, especially with inputs, e.g.,
`object` argument in `find_all_markers` should be more specific than "an
object".
* Consider using the native pipe `|>` over the `magrittr` pipe.
* Should the metadata of a SingleCellExperiment be saved as a JSON instead of
CSV?
* Consider using a GitHub Action to create a Docker / Containerized version of
the app.
* Isn't data such as `grch38` already available in Bioconductor? No need to
re-export from your package.
* Why not use `split` directly on a `SingleCellExperiment` rather than
`splitByCol`?

## tests

* Unknown. Resource `EH9452` failed to load.

**Comment by whtns** [comment]
Hi Marcel, thanks for your careful review. I will address all of the points you raised in the next week or two. Can you give some advice on splitting the package? In the interest of limiting dependencies, I am sympathetic. I would imagine two or three packages for single cell processing and shiny and/or plotting. How would that work with BioConductor's workflow? Would I submit each package in a separate issue? Would this review process restart? Thanks,

**Comment by LiNk-NY** [comment]
I would recommend splitting it along clusters of functionality, e.g., one package for visualization and related dependencies, one for the shiny app, and one for the data analysis, etc.
This modularity helps with stability across the workflow. Since we have already taken a look at the code, you could include the packages in the same review as an additional package comment.

**Comment by whtns** [comment]
Hi Marcel, I split the package into the following repos:
https://github.com/whtns/chevreulProcess
https://github.com/whtns/chevreulPlot
https://github.com/whtns/chevreulShiny

please see responses below, thanks!

> Hi @whtns Thank you for your submission. Please see the review below.
> 
> Best regards, Marcel
> # chevreul #3332
> 
>     * Overall the package contains a lot of functionality and as a consequence,
>       dependencies. It would be better to split the package into possibly two or
>       three separate packages that minimize the number of dependencies in each.
split into three packages; listed at top 

>     * Consider the `alabaster` framework for exporting `SingleCellExperiment`
>       and other objects in a language-agnostic format.
> 
> 
> ## DESCRIPTION
> 
>     * Is it possible to cut down some of the dependencies? The number of imports
>       can make the package fragile.
reduced dependencies after splitting
> 
>     * You may also want to split the package into separate packages (e.g., shiny
>       and visualization packages)
split into three packages; repos listed at top 
> 
> ## NAMESPACE
> 
>     * Looks good.
> 
>     * Consider removing the `%>%` dependency.
removed magrittr pipe dependency
> 
>     * Use more descriptive function names, e.g., `plot_feature`, `plot_readcount`,
>       `plot_var`, etc.
made function names more descriptive
> 
> ## vignettes
> 
>     * Use more descriptive titles in the vignette / index entry.
> 
>     * Remove `Sys.time` calls from vignette. These will be provided by the BBS.
removed from vignette
>     * (optional) Consider wrapping text longer than 80 column width.
> 
>     * Although a good reminder, it is not necessary to display
>       `BiocManager::valid()` for the user.
removed from vignette
> 
>     * The `human_gene_transcript_sce()` function in `chevreuldata` does not seem to
>       be working. I get some "resource removed" errors.
> 
>     * The code is a bit unclear. You take the `chevreul_sce` and reconstruct
>       another `SingleCellExperiment`?
removed sections to be more clear; split vignettes 
> 
>     * The code for creating the vignette is not needed. Please remove.
removed 

>     * Avoid static images of plots and provide the code needed to plot. Consider
>       moving plotting functionality into a separate package.
moved plotting functionality into chevreulPlot
>     * Include code to reproduce plots and analyses. Do not use static images.
removed all static images from vignettes; would images of app interface be useful?
> 
> ## R
> 
>     * Properly reference data objects in `transcripts_to_genes` etc. as detailed
>       in: https://contributions.bioconductor.org/data.html?q=data#exported-data-and-the-data-directory
thanks, fixed  
>     * Minimize the code repitition by extracting the common operations in both
>       conditions, e.g. in `chevreul_app.R` where
>       `featureType %in% c("transcript", "gene")`
sorry I'm not sure what this refers to

>     * Based on the `shiny` docs, `moduleServer()` is preferred over `callModule`.
>       Please update accordingly.
fixed
> 
>     * Be more descriptive in the documentation, especially with inputs, e.g.,
>       `object` argument in `find_all_markers` should be more specific than "an
>       object".
cleaned up roxygen documentation
>     * Consider using the native pipe `|>` over the `magrittr` pipe.
removed magrittr pipe dependency 
>     * Should the metadata of a SingleCellExperiment be saved as a JSON instead of
>       CSV?
I agree this is confusing, I updated relevant  functions to refer to colData rather than metadata
> 
>     * Consider using a GitHub Action to create a Docker / Containerized version of
>       the app.
> 
>     * Isn't data such as `grch38` already available in Bioconductor? No need to
>       re-export from your package.
I thought I could limit dependencies by including a table of gene metadata with coordinates and symbols. I am duplicating code from the github package annotables https://github.com/stephenturner/annotables.
> 
>     * Why not use `split` directly on a `SingleCellExperiment` rather than
>       `splitByCol`?
There doesn't seem to be a straightforward way to split an sce into a list of sces by colData values see https://github.com/drisso/SingleCellExperiment/issues/55
> 
> 
> ## tests
> 
>     * Unknown. Resource `EH9452` failed to load.
I will resolve this ExperimentHub issue. It seems useful to load a full dataset to understand all package functionality 
rather than the minimal objects I use for examples.

**Comment by LiNk-NY** [comment]
Hi Kevin, @whtns 

> would images of app interface be useful?

For the shiny app, it is ok to provide static images.

> sorry I'm not sure what this refers to

You can minimize repetitive coding by setting only the values that change e.g., 

```r
switch(featureType, transcript = "enstxp", gene = "symbol")
```

The rest of the operations are largely the same across both conditions. 


> * Consider the `alabaster` framework for exporting `SingleCellExperiment`
>   and other objects in a language-agnostic format.

Have you taken a look at this framework? It could reduce the amount of code you have to maintain. 

Best,
Marcel

**Comment by whtns** [comment]
Thanks @LiNk-NY. Responses to your most recent comments below. Is it possible to begin checking the constituent r packages via github acitons?

https://github.com/whtns/chevreulProcess
https://github.com/whtns/chevreulPlot
https://github.com/whtns/chevreulShiny 

> Hi Kevin, @whtns
> 
> > would images of app interface be useful?
> 
> For the shiny app, it is ok to provide static images.
I added a vignette to the shiny app package with static images to illustrate the interface
> > sorry I'm not sure what this refers to
> 
> You can minimize repetitive coding by setting only the values that change e.g.,
> 
> ```r
> switch(featureType, transcript = "enstxp", gene = "symbol")
> ```
I tried to correct simple if-else blocks with switch in all three packages. Is that what you had in mind?
> The rest of the operations are largely the same across both conditions.
> 
> > * Consider the `alabaster` framework for exporting `SingleCellExperiment`
> >   and other objects in a language-agnostic format.
> 
> Have you taken a look at this framework? It could reduce the amount of code you have to maintain.
The framework does seem useful. Much of the package is meant for a loaded SingleCellExperiment rather than needing a file path for upload. I've added an alabaster dependency to the shiny app where a file path is taken as an argument. 

> Best, Marcel

**Comment by LiNk-NY** [comment]
Hi Kevin, @whtns 

Can you reduce the size of the `chevreul` / `chevreulPlot` and others `git` repositories? They are over 700 MB! 

```
Receiving objects: 100% (7861/7861), 728.64 MiB | 8.82 MiB/s, done.
```


> https://github.com/whtns/chevreulProcess
> https://github.com/whtns/chevreulPlot
> https://github.com/whtns/chevreulShiny
 

You should declare these packages with 

```
AdditionalPackage: <URL>
```

I spotted some examples where your arguments are not fully listed in the default.
The function should have both options in the default so that they are visible to the
user. You should also have a `switch` function that respects the users `edb` input, if any:

```r
plot_gene_coverage_by_var <- function(
    organism = c("human", "mouse"),
    ...
) {
organism <- match.arg(organism)
edb <- switch(
    organism,
    mouse = EnsDb.Mmusculus.v79, 
    human = EnsDb.Hsapiens.v86,
    edb
)
}
```
<sub> from `chevreulPlot/R/wiggleplotr.R` </sub>

**Comment by whtns** [comment]
Thank you for this guidance. I'm declaring these additional packages below.

AdditionalPackage: https://github.com/whtns/chevreulProcess
AdditionalPackage: https://github.com/whtns/chevreulPlot
AdditionalPackage: https://github.com/whtns/chevreulShiny

I made the change to user input edb that you pointed out. I will review for other similar instances.

**Comment by LiNk-NY** [comment]
Hi Kevin, @whtns 
You might have to include only one `AdditionalPackage: <URL>` line per comment, 
i.e., three separate comments in the issue.

**Comment by whtns** [comment]
Thanks!
AdditionalPackage: https://github.com/whtns/chevreulPlot

**Comment by whtns** [comment]
AdditionalPackage: https://github.com/whtns/chevreulShiny

**Comment by whtns** [comment]
Hi Marcel @LiNk-NY ,
I split this work into three packages based on your review https://github.com/Bioconductor/Contributions/issues/3332#issuecomment-2386358040. I transferred these split packages to a new url under organizational ownership. Do I need to add again as AdditionalPackages?
https://github.com/cobriniklab/chevreulProcess
https://github.com/cobriniklab/chevreulPlot
https://github.com/cobriniklab/chevreulShiny

I am preparing to submit this work as a manuscript. Do you have any estimates on time to restart review? I am considering whether to drop references to bioconductor in the text of my manuscript.

**Comment by whtns** [comment]
Hi @lshep, 
I added an additional dataset to the `chevreuldata` experimenthub package and transferred to organizational ownership. Should I add as an additional package? 
https://github.com/cobriniklab/chevreuldata

**Comment by LiNk-NY** [comment]
Hi Kevin @whtns 
I think we can get these packages in once they are built on the SPB. 
Lori, @lshep, can get them added to the auto-build step.



> transferred to organizational ownership. Should I add as an additional package?

Yes, I think you would have to re-comment packages with the `AdditionalPackage` key.

**Comment by whtns** [comment]
Thanks! 
AdditionalPackage: https://github.com/cobriniklab/chevreulProcess

**Comment by whtns** [comment]
Hi @LiNk-NY and @lshep, when I attempted to add packages with the new urls this issue was closed. Can you let me know how to proceed?

**Comment by LiNk-NY** [comment]
This is likely a bug in the duplicate `AdditionalComments` part of the SPB code. 
I think Lori should have a fix for it / override when she gets back from her break. 

Best,
Marcel

**Comment by lshep** [comment]
its not a bug --  we don't allow duplicate submissions and the package was already submitted here but it seems with a different github account/username.  We do this purposely so that multiple labs don't submit the same issue and/or to prevent two packages from being submitted with the same name from different labs as all package names must be unique 
packages/additional packages were submitted from whtns  is that correct or should it be from cobriniklab --  I'm trying to make sense of it all; are you proceeding on this issue or are you opening a new issue ?

**Comment by LiNk-NY** [comment]
If possible, we could proceed on this issue. 
I think @whtns wants to proceed from the cobiniklab repos. 

@whtns FWIW, the repos on your accounts should be forks of the cobiniklab repos.

**Comment by lshep** [comment]
So there are 5 packages total now for this issue
 https://github.com/cobriniklab/chevreul
https://github.com/cobriniklab/chevreuldata
https://github.com/cobriniklab/chevreulProcess
https://github.com/cobriniklab/chevreulPlot
https://github.com/cobriniklab/chevreulShiny


does chevreul and chevreuldata need to be reset on our git.bioconductor.org or do they have consistent git histories that can still be pushed too?

**Comment by whtns** [comment]
Thank you, yes, to aid paper submission I'd like to proceed with these packages under the cobriniklab organization. I now understand that this process might have been simpler if I set up forks between my account and the organization. Thanks for your help despite my inexperience.

> So there are 5 packages total now for this issue https://github.com/cobriniklab/chevreul https://github.com/cobriniklab/chevreuldata https://github.com/cobriniklab/chevreulProcess https://github.com/cobriniklab/chevreulPlot https://github.com/cobriniklab/chevreulShiny
> 
> does chevreul and chevreuldata need to be reset on our git.bioconductor.org or do they have consistent git histories that can still be pushed too?

Chevreul does not have a consistent git history. It was formerly a single submission but after splitting into the three chevreul* packages is now a meta-package. I don't know if Bioconductor encourages/allows this approach. 

chevreuldata should have a consistent git history but I have a separate concern with the package, quoted below 

> Hi @lshep, I added an additional dataset to the `chevreuldata` experimenthub package and transferred to organizational ownership. Should I add as an additional package? https://github.com/cobriniklab/chevreuldata

Thanks for all your help

**Comment by lshep** [comment]
Ok...  to be on the safe side... I'll reset chevreuldata and chevreul on git.bioconductor.org.  I won't have time to do it today but likely tomorrow morning.  In the meantime ... to be safe... I'm going to reset the submission queue entry database to make sure we process the correct repo.... can you please do an individual AdditionalPackage comments for the three additional....

**Comment by whtns** [comment]
Thanks for your help @lshep 
adding AdditionalPackage comments below 
AdditionalPackage: https://github.com/cobriniklab/chevreulProcess

**Comment by whtns** [comment]
AdditionalPackage: https://github.com/cobriniklab/chevreulPlot

**Comment by whtns** [comment]
AdditionalPackage: https://github.com/cobriniklab/chevreulShiny

**Comment by lshep** [comment]
Please don't include remotes in the description.  The BBS doesn't respect remotes and it should come from Bioconductor.  By submitting under the same issue the issue will be able to find the relevant packages and then they should be listed appropriately in Suggest/Imports/Depends

**Comment by lshep** [comment]
We cannot currently process chevreulPlot as there are files over 5Mb 
```
Error in .precheck_filesize(wd) : files larger than 5Mb:
  /inst/extdata/ds20181001-0001_Aligned.sortedByCoord.out.bw
  /inst/extdata/ds20181001-0173_Aligned.sortedByCoord.out.bw
```

**Comment by whtns** [comment]
Thanks @lshep, I've changed default branches for all repositories to devel. There should now be no remotes listed in DESCRIPTIONS. I've also removed any files larger than 5Mb

**Comment by lshep** [comment]
sorry since we switched builders its not picking up the data package.  If you do a version bump for the data package and have it build, the other packages will be able to find it on rebuild

**Comment by lshep** [comment]
I'm not sure why this is happening and it keeps resending the build. Looking into it.

**Comment by whtns** [comment]
Hi Marcel @LiNk-NY and Lori @lshep

Three packages referenced in this issue have passed R CMD check without WARNING or ERROR:
1. chevreulProcess
2. chevreulPlot
3. chevreulShiny

Two other packages chevreuldata and chevreul have not passed. 
The ERROR in chevreuldata is due to TIMEOUT during example checking. I'd like to drop this package from the issue and will drop any reference in package vignettes
The WARNING in chevreul is due to. "No Bioconductor dependencies detected..." I don't know how to resolve this WARNING as this is a metapackage and should have dependencies only on the three passing packages listed above. 

Could you let me know how to proceed? Thank you!

**Comment by LiNk-NY** [comment]
Hi Kevin, @whtns 
Sorry for the delay. I was out of town. 
Can you please bump the version of the relevant packages? 
Thank you!

**Comment by whtns** [comment]
Hi Marcel, @LiNk-NY. I updated package versions for the three separate packages that you suggested, but no longer meet minimum checktime requirements. Could this be due to changes in minimum R and Bioconductor versions? Do you have information on documentation to meet the 10 min threshold?

**Comment by whtns** [comment]
Hi Marcel, @LiNk-NY. I updated package versions for the three separate packages that you suggested

1. chevreulProcess
2. chevreulPlot
3. chevreulShiny

all passed without WARNING or ERROR. Can you attempt review again?

**Comment by whtns** [comment]
Hello Marcel @LiNk-NY, I just wanted to follow up on this request for review. The three packages involved in this issue all pass R CMD check without WARNING or ERROR.

**Comment by LiNk-NY** [comment]
> Hello Marcel @LiNk-NY, I just wanted to follow up on this request for review. The three packages involved in this issue all pass R CMD check without WARNING or ERROR.

Hi @whtns 
Sorry for the delay. I will take a look tomorrow.

**Comment by LiNk-NY** [comment]
Hi @whtns 
Can you bump the version for `chevreulProcess`?

**Comment by whtns** [comment]
updated! @LiNk-NY

**Comment by LiNk-NY** [comment]
> The function should have both options in the default so that they are visible to the
> user. You should also have a `switch` function that respects the users `edb` input, if any:

This wasn't improved. Please include `c("human", "mouse")` in the argument default. If `edb` is not `missing` use that instead of the `switch` operation:

```r
plot_gene_coverage_by_var <- function(
    organism = c("human", "mouse"),
    ...
) {
organism <- match.arg(organism)
if (missing(edb))
    edb <- switch(
        organism,
        mouse = EnsDb.Mmusculus.v79, 
        human = EnsDb.Hsapiens.v86
    )
}
```

**Comment by whtns** [comment]
Thank you @LiNk-NY, I've updated the `plot_gene_coverage_by_var` function and rebuilt without ERROR or WARNING

**Comment by LiNk-NY** [comment]
Hi @whtns 
Please make sure that there are no mentions of `chevreul` in files, e.g. `chevreulShiny/tests/testthat.R` etc.

**Comment by whtns** [comment]
Hi @LiNk-NY, I replaced references to "chevreul" in `chevreulShiny` and `chevreulPlot`. I'm not seeing the specific issue you [refer to](https://github.com/cobriniklab/chevreulShiny/blob/devel/tests/testthat.R) in chevreulShiny/tests/testthat.R. Can you take another look? Thanks

**Comment by whtns** [comment]
Hi @LiNk-NY , just refreshing this thread. Can you elaborate on your comment from last week? Thanks!

**Comment by LiNk-NY** [comment]
Hi Kevin, @whtns 

Thanks, I was looking at an old checkout of `whtns/chevreulShiny`. 

There are still some unresolved issues in the packages. 
Please make sure you are running build and check after making changes. 

Make sure you are adding these packages to the `Imports` field in the `DESCRIPTION` in `chevreulShiny`: 

```r
mouse = EnsDb.Mmusculus.v79,
human = EnsDb.Hsapiens.v86
```

and use the `::` double colon to reference them, e.g., `EnsDb.Mmusculus.v79::EnsDb.Mmusculus.v79`. 

_Alternatively_ (preferred), you may move them to the `Suggests` field and add conditional code to check if the package is available.
See here for an example: https://contributions.bioconductor.org/description.html?q=Suggests#depends-imports-suggests-enhances

Resolve this warning in `chevreulProcess` : 

```r
* checking DESCRIPTION meta-information ... WARNING
Invalid license file pointers: LICENSE
```

The LICENSE file cannot have an `.md` extension for `R CMD check`. 

See the warning in `chevreulPlot`:

```r
* checking for code/documentation mismatches ... WARNING
Codoc mismatches from Rd file 'plot_gene_coverage_by_var.Rd':
plot_gene_coverage_by_var
  Code: function(genes_of_interest = "NRL", cell_metadata, bigwig_tbl,
                 group_by = "batch", values_of_interest = NULL,
                 organism = c("human", "mouse"), edb = NULL, heights =
                 c(3, 1), scale_y = "log10", reverse_x = FALSE, start =
                 NULL, end = NULL, summarize_transcripts = FALSE, ...)
  Docs: function(genes_of_interest = "NRL", cell_metadata, bigwig_tbl,
                 group_by = "batch", values_of_interest = NULL,
                 organism = "human", edb = NULL, heights = c(3, 1),
                 scale_y = "log10", reverse_x = FALSE, start = NULL,
                 end = NULL, summarize_transcripts = FALSE, ...)
  Mismatches in argument default values:
    Name: 'organism' Code: c("human", "mouse") Docs: "human"
```



Best,
Marcel

**Comment by whtns** [comment]
Hi @LiNk-NY, 

I've responded to this questions that you raised on 1/24. All packages pass R CMD check without warnings or errors as shown in the previous three entries in this thread. 

> Make sure you are adding these packages to the `Imports` field in the `DESCRIPTION` in `chevreulShiny`:
> 
> mouse = EnsDb.Mmusculus.v79,
> human = EnsDb.Hsapiens.v86
> and use the `::` double colon to reference them, e.g., `EnsDb.Mmusculus.v79::EnsDb.Mmusculus.v79`.
> 
> _Alternatively_ (preferred), you may move them to the `Suggests` field and add conditional code to check if the package is available. See here for an example: https://contributions.bioconductor.org/description.html?q=Suggests#depends-imports-suggests-enhances

I have added annotation packages to `Suggests` as you outlined
> 
> Resolve this warning in `chevreulProcess` :
> 
> * checking DESCRIPTION meta-information ... WARNING
> Invalid license file pointers: LICENSE
> The LICENSE file cannot have an `.md` extension for `R CMD check`.

I have removed duplicate LICENSE.md files from all packages

> See the warning in `chevreulPlot`:
> 
> * checking for code/documentation mismatches ... WARNING
> Codoc mismatches from Rd file 'plot_gene_coverage_by_var.Rd':
> plot_gene_coverage_by_var
>   Code: function(genes_of_interest = "NRL", cell_metadata, bigwig_tbl,
>                  group_by = "batch", values_of_interest = NULL,
>                  organism = c("human", "mouse"), edb = NULL, heights =
>                  c(3, 1), scale_y = "log10", reverse_x = FALSE, start =
>                  NULL, end = NULL, summarize_transcripts = FALSE, ...)
>   Docs: function(genes_of_interest = "NRL", cell_metadata, bigwig_tbl,
>                  group_by = "batch", values_of_interest = NULL,
>                  organism = "human", edb = NULL, heights = c(3, 1),
>                  scale_y = "log10", reverse_x = FALSE, start = NULL,
>                  end = NULL, summarize_transcripts = FALSE, ...)
>   Mismatches in argument default values:
>     Name: 'organism' Code: c("human", "mouse") Docs: "human"

I've updated documentation in chevreulPlot for plot_gene_coverage_by_var

> Best, Marcel

Thanks!

**Comment by LiNk-NY** [comment]
Thanks for those updates. The packages have been accepted. 
Best regards,
Marcel

**Comment by lshep** [comment]
Can I just confirm that all the following packages should be included/are accepted: 
chevreul
chevreuldata
chevreulPlot
chevreulProcess
chevreulShiny

**Comment by whtns** [comment]
Thanks @lshep, see my [prior comment](https://github.com/Bioconductor/Contributions/issues/3332#issuecomment-2486860465) I had trouble with the meta package, chevreul, as it lacks bioconductor dependencies and with the experimenthub package, chevreuldata, due to an inscrutable timeout during example checks. I will drop chevreuldata and forgo chevreul if neither of these concerns can be resolved.

**Comment by lshep** [comment]
Your packages have been added to Bioconductor devel branch. 
See further instructions at

https://bioconductor.org/developers/how-to/git/

for working with this repository. See especially

https://bioconductor.org/developers/how-to/git/new-package-workflow/
https://bioconductor.org/developers/how-to/git/sync-existing-repositories/

to keep your GitHub and Bioconductor repositories in sync.

Your package will be included in the next nigthly 'devel' build (check-out from git at about 6 pm Eastern; build completion around 2pm Eastern the next day) at

https://bioconductor.org/checkResults/

(Builds sometimes fail, so ensure that the date stamps on the main landing page are consistent with the addition of your package). Once the package builds successfully, you package will be available for download in the 'Devel' version of Bioconductor using `BiocManager::install("chevreulPlot")`. The package 'landing page' will be created at

https://bioconductor.org/packages/chevreulPlot
https://bioconductor.org/packages/chevreulProcess
https://bioconductor.org/packages/chevreulShiny



If you have any questions, please contact the bioc-devel mailing list (https://stat.ethz.ch/mailman/listinfo/bioc-devel); this issue will not be monitored further.

[Add an SSH key to your github account]: https://help.github.com/articles/connecting-to-github-with-ssh/
[Submit your SSH key to Bioconductor]: https://git.bioconductor.org/BiocCredentials/

**Comment by lshep** [comment]
@whtns  Hello, I am doing some ExperimentHub clean up.  Are the ExperimentHub records from chevreuldata referenced in any of
```
https://bioconductor.org/packages/chevreulPlot
https://bioconductor.org/packages/chevreulProcess
https://bioconductor.org/packages/chevreulShiny
```
Else we will remove the entry from ExperimentHub and delete the files off our hosted server.  Data hosted in Hubs must be associated with a submitted/accepted package.

**Comment by lshep** [comment]
@whtns   Please note your NEWS files are not formatted properly.  As these are new submissions we will change to be "Initial Bioconductor submission" on our release announcement but to be included in future release announcements you will have to re-format.        Also please see the inquiry that was left unanswered above

**Comment by whtns** [comment]
Thanks, sorry I missed the prior inquiry above. The ExperimentHub package chevreuldata uses a dataset stored on a lab server and not in the Microsoft Azure Genomic Data Lake. The associated packages listed above do not depend on the chevreuldata package right now. I will re-format the NEWS files.

**Comment by lshep** [comment]
@whtns with chevreuldata not being submitted to Bioconductor as an official package we will be removing the entry in the ExperimentHub unless it is submitted and adheres to Bioconductor standards

**Comment by whtns** [comment]
thanks for notifying me
